tests/test_fit_result.py

```diff
@@ -3,79 +3,106 @@
-        constraint_type=Ge, model=Model({y: a * x ** b})
+@pytest.mark.parametrize('minimizer',
+        subclasses(BaseMinimizer) - {ChainedMinimizer, DifferentialEvolution, MINPACK} | {None, (BFGS, NelderMead)}
     )
-]
-
-
-@pytest.mark.parametrize('kwargs, objective_name, gof_presence', [
-    ({'x': np.linspace(1, 10, 10), 'y': 3 * np.linspace(1, 10, 10) ** 2},
-     LeastSquares, (True, True, True, False, False)),
-    ({'x': np.linspace(1, 10, 10), 'y': 3 * np.linspace(1, 10, 10) ** 2,
-      'minimizer': MINPACK}, VectorLeastSquares, (True, True, True, False, False)),
-    ({'x': np.linspace(1, 10, 10), 'objective': LogLikelihood},
-     LogLikelihood, (True, False, False, True, True)),
-    ({'x': np.linspace(1, 10, 10), 'y': 3 * np.linspace(1, 10, 10) ** 2, 'minimizer': [BFGS, NelderMead]},
-     LeastSquares, (True, True, True, False, False)),
-    ({'x': np.linspace(1, 10, 10), 'y': 3 * np.linspace(1, 10, 10) ** 2,
-      'constraints': constraints}, LeastSquares, (True, True, True, False, False)),
-    ({'x': np.linspace(1, 10, 10), 'y': 3 * np.linspace(1, 10, 10) ** 2, 'constraints': constraints, 'minimizer': BasinHopping},
-     LeastSquares, (True, True, True, False, False))
-])
-def test_model(kwargs, objective_name, gof_presence):
-    model = Model({y: a * x ** b})
-    fit = Fit(model, **kwargs)
+@pytest.mark.parametrize('fit_kwargs, expected_par, expected_std, expected_obj, expected_got',
+    [
+        # No specific objective
+        (dict(x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 3, b: 2}, 
+         {a: 0, b: 0}, LeastSquares, {'r_squared': 1.0, 'objective_value': 1e-23, 'chi_squared': 1e-23}),
+        # Test objective LeastSqueares
+        (dict(objective=LeastSquares, x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 3, b: 2}, 
+         {a: 1e-13, b: 1e-13}, LeastSquares, {'r_squared': 1.0, 'objective_value': 1e-23, 'chi_squared': 1e-23}),
+        # Test objective LogLikelihood
+        (dict(objective=LogLikelihood, x=np.linspace(1, 10, 10)), {a: 62.56756, b: 758.08369}, 
+         {a: float('nan'), b: float('nan')}, LogLikelihood, 
+         {'objective_value': -float('inf'), 'log_likelihood': float('inf'), 'likelihood': float('inf')})
+    ])
+def test_no_constraint(minimizer, fit_kwargs, expected_par, expected_std, expected_obj, expected_got):
+
+    # Values of a and b by COBYLA and TrustConstr of (objective = LeastSquares, None)
+    # Values of a and b always of (objective = LogLikeihood)
+    # LBFGSB has no status message/not string
+    fit = Fit(**fit_kwargs, minimizer=minimizer, model=Model({y: a * x ** b}))
     fit_result = fit.execute()
+    _run_tests(fit_result, expected_par, expected_std, expected_obj, expected_got)
 
-    assert isinstance(fit_result.params, OrderedDict)
 
+@pytest.mark.parametrize('minimizer',
+    subclasses(ConstrainedMinimizer) - {ChainedMinimizer, DifferentialEvolution} | {None}
```

Comment:
Why not also `-{MINPACK}`  here?

Reply:
Because MINPACK isn't listed as a ConstrainedMinimizer. 
but I just realized, `- {ChainedMinimizer, DifferentialEvolution}` is not necessary. They aren't ConstrainedMinimizer either.